content/blog/survival-analysis-parsnip-adjacent/index.Rmd

```diff
@@ -0,0 +1,227 @@
+
+As part of the modelling workflow, we may want to use these predictions with a performance metric such as concordance. This metric captures the probability that the prediction goes in the same direction as the actual data. While the data captures _survival_ time, the linear predictor for a proportional hazards model is a factor on the _risk_. The predicted survival time is longer if the linear predictor is lower. To make the prediction and the observation go in the same direction, we flip the sign of the prediction. This may be opposite to the sign of predictions obtained using the underlying engine directly.
+
+```{r, lp-sign}
+fit_censored <- proportional_hazards() %>% 
+  set_engine("survival") %>% 
+  fit(Surv(time, status) ~ X1 + X2, data = train_dat)
+pred_censored <- predict(fit_censored, type = "linear_pred", new_data = test_pred)
+
+pred <- bind_cols(
+  pred,
+  pred_censored %>% rename(pred_censored = .pred_linear_pred)
+)
+
+pred %>% 
+  ggplot(aes(linear_pred, pred_censored)) + 
+  geom_point() +
+  geom_abline(slope = -1, intercept = 0)
+```
+
+  
+
+### Model and prediction types available in censored
+
+Currently, censored contains the following models, engine, and prediction types with `r cli::symbol$tick` indicating "available", `r cli::symbol$cross` indicating "unavailable", and `r cli::symbol$circle` indicating "to do".
+
+```{r table, echo = FALSE, results = "asis"}
+yep <- cli::symbol$tick
+nope <- cli::symbol$cross
+todo <- cli::symbol$circle
+
+mod_names <- get_from_env("models")
+model_info <-
+  map_dfr(mod_names, ~ get_from_env(paste0(.x, "_predict")) %>% mutate(alias = .x))
+
+done <- model_info %>%
+  filter(mode == "censored regression") %>%
+  select(model = alias, engine, type, mode) %>% 
+  pivot_wider(names_from = type, 
+              values_from = mode, 
+              values_fill = nope, 
+              values_fn = function(x) yep) %>% 
+  mutate(linear_pred = case_when(model == "survival_reg" ~ todo,
+                                 TRUE ~ linear_pred),
+         time = case_when(time == nope ~ todo,
+                          TRUE ~ time)
+  )
+to_do <- tribble(
+  ~model, ~engine, ~survival, ~linear_pred, ~time, ~raw, ~quantile, ~hazard,
+  "rand_forest", "ranger", todo, nope, todo, nope, nope, nope,
+  "rand_forest", "randomForestSRC",  todo, nope, todo, nope, nope, nope
+)
+
+bind_rows(done, to_do) %>% 
+  arrange(model, engine) %>% 
+  knitr::kable(escape = FALSE) 
+```
+
+
+
+### What's next?
+
+We want survival analysis to be a first-class citizen in tidymodels which means having access to preprocessing, resampling, models, metrics, and tuning. The next steps on that path are a role for censoring indicator columns and a step `step_surv()` in recipes as well as a new ROC metric for survival models in yardstick. An adaption of workflows and tune is to follow after that. Additionally, we are in the process of figuring out how different types of censoring and time-dependent effects fit into our API.
```

Comment:
I'd make the next steps into a bulleted list to make them visually stand out more. Maybe start GH issues and like to the issues so that people can provide direct feedback easily.